core/src/test/java/org/elasticsearch/indices/state/OpenCloseIndexIT.java

```diff
@@ -190,6 +190,39 @@ public void testCloseOpenAllWildcard() {
         assertThat(openIndexResponse.isAcknowledged(), equalTo(true));
         assertIndexIsOpened("test1", "test2", "test3");
     }
+    
+    // if there are no indices to open/close throw an exception
+    public void testOpenCloseWildCardsNoIdcicesDefault() {
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("_all").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("_all").execute().actionGet());
+    }
+    
+    // if there are no indices to open/close and allow_no_indices=true, the open/close is a no-op
+    public void testOpenCloseWildCardsNoIdcicesAllowNoIndices() {
+        IndicesOptions openIncidesOptions = IndicesOptions.fromOptions(false, true, false, true);
+        IndicesOptions closeIncidesOptions = IndicesOptions.fromOptions(false, true, true, false);
+
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test").execute().actionGet());
+        
+        assertAcked(client().admin().indices().prepareClose("test*").setIndicesOptions(openIncidesOptions).execute().actionGet());
+        assertAcked(client().admin().indices().prepareClose("test*").setIndicesOptions(closeIncidesOptions).execute().actionGet());
+        
+        assertAcked(client().admin().indices().prepareClose("*").setIndicesOptions(openIncidesOptions).execute().actionGet());
+        assertAcked(client().admin().indices().prepareClose("*").setIndicesOptions(closeIncidesOptions).execute().actionGet());
+        
+        assertAcked(client().admin().indices().prepareClose("_all").setIndicesOptions(openIncidesOptions).execute().actionGet());
+        assertAcked(client().admin().indices().prepareClose("_all").setIndicesOptions(closeIncidesOptions).execute().actionGet());
```

Comment:
if you want to save some characters you can replace all these `execute().actionGet()` with just `get()`